### DISCUSSION TITLE
Can Unicode texts be written to a .mid file?

### DISCUSSION THREAD

**Discussion by Jenix-Park** [melanchall/drywetmidi#184] [Q&A] [answered]
I found a [TextEncoding option](https://melanchall.github.io/drywetmidi/api/Melanchall.DryWetMidi.Core.WritingSettings.html#Melanchall_DryWetMidi_Core_WritingSettings_TextEncoding) in the WritingSettings class. It's no surprise that the default is ASCII because the MIDI standard only supports ASCII texts.
But what will happen when we switch the encoding to something like UTF-8 for our texts written in multilanguage?
Can they be seen via any other MIDI editors/players correctly without any corruption?

**Comment by melanchall** [comment, marked as answer]
> Can they be seen via any other MIDI editors/players correctly without any corruption?

No, that is not guaranteed. You can specify `TextEncoding` property on `WritingSettings`/`ReadingSettings` to write/read MIDI data by DryWetMIDI.

Also some DAWs can write MIDI with text events where text is stored in Unicode to save, for example, lyrics in your local language. And DryWetMIDI allows you correctly read such texts via specifying `TextEncoding` property. But again, there is no guarantee that such text will be correctly read by other software.
